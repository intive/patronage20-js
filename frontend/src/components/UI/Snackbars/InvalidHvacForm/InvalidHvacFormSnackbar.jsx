import { useEffect } from 'react'
import { useSelector } from 'react-redux'
import { useSnackbar } from 'notistack'
import { useTranslation } from 'react-i18next'

export default function InvalidHvacFormSnackbar () {
  const { t } = useTranslation()
  const { enqueueSnackbar, closeSnackbar } = useSnackbar()
  const hvacRoomsValidForm = useSelector((state) => state.sensor.hvacRoomsValidForm)

  useEffect(() => {
    if (!hvacRoomsValidForm) {
      enqueueSnackbar(t('hvac:invalid-form'), {
        variant: 'warning',
        key: 'invalid-form-snackbar'
      })
    } else if (hvacRoomsValidForm) {
      closeSnackbar('invalid-form-snackbar')
    }
  }, [hvacRoomsValidForm])

  return null
}
