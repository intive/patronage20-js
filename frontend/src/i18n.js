import { initReactI18next } from 'react-i18next'
import i18n from 'i18next'

import translations from './common/locales'

const ns = {
  namespaces: ['common', 'dashboard', 'hvac', 'authors']
}

i18n
  // pass the i18n instance to react-i18next.
  .use(initReactI18next)
  // init i18next
  // for all options read: https://www.i18next.com/overview/configuration-options
  .init({
    lng: navigator.language.slice(0, 2),
    fallbackLng: 'en',
    // debug: true,
    ns,
    defaultNS: 'common',
    resources: translations,
    react: {
      useSuspense: false
    },
    interpolation: {
      escapeValue: false // not needed for react as it escapes by default
    }
  })

export default i18n
