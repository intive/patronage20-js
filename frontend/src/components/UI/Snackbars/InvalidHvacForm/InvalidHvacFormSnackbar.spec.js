import React from 'react'
import { Provider } from 'react-redux'
import { create } from 'react-test-renderer'
import { render, screen } from '@testing-library/react'
import configureStore from 'redux-mock-store'
import { I18nextProvider } from 'react-i18next'
import { SnackbarProvider } from 'notistack'
import i18n from '../../../../i18n'
import InvalidHvacFormSnackbar from './index'

jest.mock('notistack', () => ({
  ...jest.requireActual('notistack')
}))

describe('showing <InvalidHvacFormSnackbar />', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })
  it('should show a snackbar if hvacRoomsValidForm is false', () => {
    const initialState = { sensor: { hvacRoomsValidForm: false } }
    const mockStore = configureStore()
    const store = mockStore(initialState)

    render(
      <Provider store={store}>
        <SnackbarProvider>
          <I18nextProvider i18n={i18n}>
            <InvalidHvacFormSnackbar />
          </I18nextProvider>
        </SnackbarProvider>
      </Provider>
    )
    expect(screen.getByText('INVALID FORM. Fill out the required fields.')).not.toEqual(null)
  })

  it('should close a snackbar if snackbar is open and hvacRoomsValidForm is set to true, should not open a new one', () => {
    const notistack = require('notistack')

    const closeSnackbar = jest.fn()
    const enqueueSnackbar = jest.fn()
    jest.spyOn(notistack, 'useSnackbar').mockImplementation(() => { return { enqueueSnackbar, closeSnackbar } })

    const initialState = { sensor: { hvacRoomsValidForm: true } }
    const mockStore = configureStore()
    const store = mockStore(initialState)

    render(
      <Provider store={store}>
        <SnackbarProvider>
          <InvalidHvacFormSnackbar />
        </SnackbarProvider>
      </Provider>
    )

    expect(closeSnackbar).toHaveBeenCalledTimes(1)
    expect(enqueueSnackbar).not.toHaveBeenCalled()
  })

  it('renders InvalidHvacFormSnackbar component', () => {
    const initialState = { sensor: { } }
    const mockStore = configureStore()
    const store = mockStore(initialState)

    const root = create(
      <Provider store={store}>
        <SnackbarProvider>
          <InvalidHvacFormSnackbar />
        </SnackbarProvider>
      </Provider>
    )
    expect(root.toJSON()).toMatchSnapshot()
  })
})
