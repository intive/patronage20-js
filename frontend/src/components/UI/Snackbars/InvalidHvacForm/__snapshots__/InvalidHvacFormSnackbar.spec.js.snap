// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`showing <InvalidHvacFormSnackbar /> renders InvalidHvacFormSnackbar component 1`] = `null`;
